pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleSchemaTest.java

```diff
@@ -252,4 +264,126 @@ private void getSchemaVersionFromMessages(boolean batching) throws Exception {
             assertEquals(data.getValue(), new V1Data(1));
         }
     }
+
+    @Test(dataProvider = "batchingModes")
+    public void testAutoConsume(boolean batching) throws Exception {
+        String topic = "my-property/my-ns/schema-test-auto-consume-" + batching;
+
+        try (Producer<V1Data> p = pulsarClient.newProducer(Schema.AVRO(V1Data.class))
+             .topic(topic)
+             .enableBatching(batching)
+             .create();
+             Consumer<GenericRecord> c = pulsarClient.newConsumer(Schema.AUTO_CONSUME())
+             .topic(topic)
+             .subscriptionName("sub1")
+             .subscriptionInitialPosition(SubscriptionInitialPosition.Earliest)
+             .subscribe()) {
+
+            int numMessages = 10;
+
+            for (int i = 0; i < numMessages; i++) {
+                p.sendAsync(new V1Data(i));
+            }
+            p.flush();
+
+            for (int i = 0; i < numMessages; i++) {
+                Message<GenericRecord> data = c.receive();
+                assertNotNull(data.getSchemaVersion());
+                assertEquals(data.getValue().getField("i"), i);
+            }
+        }
+    }
+
+    @Test(dataProvider = "batchingModes")
+    public void testAutoKeyValueConsume(boolean batching) throws Exception {
+        String topic = "my-property/my-ns/schema-test-auto-keyvalue-consume-" + batching;
+
+        Schema<KeyValue<V1Data, V1Data>> pojoSchema = Schema.KeyValue(
+            Schema.AVRO(V1Data.class),
+            Schema.AVRO(V1Data.class),
+            KeyValueEncodingType.SEPARATED);
+
+        try (Producer<KeyValue<V1Data, V1Data>> p = pulsarClient.newProducer(pojoSchema)
+             .topic(topic)
+             .enableBatching(batching)
+             .create();
+             Consumer<KeyValue<GenericRecord, GenericRecord>> c1 = pulsarClient.newConsumer(
+                 Schema.KeyValue(
+                     Schema.AUTO_CONSUME(),
+                     Schema.AUTO_CONSUME(),
+                     KeyValueEncodingType.SEPARATED))
+             .topic(topic)
+             .subscriptionName("sub1")
+             .subscriptionInitialPosition(SubscriptionInitialPosition.Earliest)
+             .subscribe();
+             Consumer<KeyValue<V1Data, V1Data>> c2 = pulsarClient.newConsumer(
+                 Schema.KeyValue(
+                     Schema.AVRO(V1Data.class),
+                     Schema.AVRO(V1Data.class),
+                     KeyValueEncodingType.SEPARATED))
+             .topic(topic)
+             .subscriptionName("sub2")
+             .subscriptionInitialPosition(SubscriptionInitialPosition.Earliest)
+             .subscribe();
```

Comment:
Will c2 be reject by broker? default is use exclusive subscription mode.

Reply:
No because they are using different subscription names